### PR TITLE
istioctl describe: fixes, test case, remove deprecated code

### PR DESCRIPTION
**Pull request by esnible**
Fixes `istioctl x describe` recent bug
Adds an integration test for `istioctl x describe`
Remove deprecated `istioctl get` `put` `delete` `create` `set-context`

Fixes https://github.com/istio/istio/issues/18972
Fixes https://github.com/istio/istio/issues/20157

### CODE REVIEW

File: tests/integration/pilot/istioctl_test.go [hunk 1]
```diff
@@ -72,3 +117,60 @@ func TestVersion(t *testing.T) {
 			}
 		})
 }
+
+func TestDescribe(t *testing.T) {
+	framework.NewTest(t).
+		RunParallel(func(ctx framework.TestContext) {
+
+			ctx.NewSubTest("ISTIOCTL").
```
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go]:
```suggestion
			ctx.NewSubTest("istioctl").
```

Comment (howardjohn) [reply tests/integration/pilot/istioctl_test.go]: actually just remove the subtest entirely

File: tests/integration/pilot/istioctl_test.go [hunk 2]
```diff
@@ -72,3 +117,60 @@ func TestVersion(t *testing.T) {
...
+
+					args := []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+						"x", "describe", "svc", "a"}
+					output := istioCtl.InvokeOrFail(t, args)
+					g := gomega.NewGomegaWithT(t)
+					g.Expect(output).To(gomega.BeIdenticalTo(describeSvcAOutput))
+
+					podID, err := getPodID(a)
+					if err != nil {
+						t.Fatalf("Could not get Pod ID: %v", err)
```
Comment (howardjohn) [inline tests/integration/pilot/istioctl_test.go]: this needs to be the same thing that was passed to the func the tests runs in (`ctx` in this case)

**Comment by esnible** [comment]
@howardjohn Test question.  The integration test works locally, and works for the pilot integration tests, but fails for integ-new-install-k8s-tests_istio because for that test `istioctl x describe` believes the pods are not PERMISSIVE, but DISABLE.

https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20169/integ-new-install-k8s-tests_istio/5229

I am not sure what to do.  I don't want to relax the output comparison because if it changes I want this test to pick it up.  I don't know how to recreate locally the options used for integ-new-install-k8s-tests_istio, or why Pilot's /debug/authenticationz?proxyID will return different values when I use it.

File: tests/integration/pilot/istioctl_test.go (hunk 2 above)
Comment (esnible) [reply tests/integration/pilot/istioctl_test.go]: I don't understand the comment.  What is the problem?

Comment (howardjohn) [reply tests/integration/pilot/istioctl_test.go]: See the test failure: `testing.go:864: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test`

You can't use the parents `t`, its not a valid use of the go testing library

**Comment by howardjohn** [comment]
@esnible that test is just running `istioctl manifest apply`, not sure why it would be different? At this point (I thought, but this makes me think I am wrong) its almost entirely redundant from the pilot-k8s-test, we should probably clean it up - I am curious why its different here

**Comment by ayj** [comment]
LGTM overall. Will add final approval after John's test comments are addressed.

**Comment by esnible** [comment]
@ayj I have completed the things @howardjohn asked for.  Please review now.

**Comment by howardjohn** [comment]
This is the only reason we have a fair amount of legacy junk in Pilot, will be useful to get rid of this

**Comment by howardjohn** [comment]
@nmittler can you take a look? Then we can clean up the config store a bit

**Comment by esnible** [comment]
@howardjohn What is needed to proceed on this?

**Comment by howardjohn** [comment]
@esnible  I am pretty sure the code changes were merged but somehow the PR is open, so  Ithink we can just close this?

**Comment by howardjohn** [comment]
Just verified the changes are in, so I recommend just closing

**Comment by nmittler** [comment]
@esnible @howardjohn do we know how these changes got committed? Maybe someone force pushed from the command-line to the master branch accidentally? I wouldn't expect that would even be possible.

**Comment by howardjohn** [comment]
It was actually prow that merged based on the commit log. Apparently the
prow logs looked normal so the assumption it was a github issue. The same
happened to my PR as well and I definitely didn't force push since I wasn't
around when it happened

**Comment by fejta** [comment]
Please rebase this PR. It looks like github only merged some of the commits (3 of 13)?

**Comment by esnible** [comment]
Closing because this is already merged!

https://github.com/istio/istio/commit/926a1bfd67c8daa42651d9cd60ee94820672c67a#diff-f4fb5eb2363cca54308accde8f31be9c is the commit.

**Pull request closed without merging**